src/transformStateWithClones.js

```
@@ -7,7 +7,44 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newArray = [];
+
+  const copyState = { ...state };
+
+  actions.forEach(action => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(copyState, action.extraData);
+
+        break;
+
+      case 'removeProperties':
+        const removeKeys = action.keysToRemove.values();
+
+        for (const key of removeKeys) {
+          delete copyState[key];
+        }
+
+        break;
+
+      case 'clear':
+
+        for (const key in copyState) {
+          delete copyState[key];
+        }
+
+        break;
+
+      default:
+        return state;
+    }
```

It would be great to better throw an error instead of returning state